gw_eccentricity/eccDefinition.py

```diff
@@ -579,6 +619,22 @@ def process_data_dict(self,
                 "1. 'hlm' OR \n"
                 "2. 'amplm' and 'phaselm'\n"
                 "But not both 1. and 2. at the same time."))
+        # if the system is precessing and the modes are given in the inertial
+        # frame, rotate the modes and obtain the corresponding modes in the
+        # coprecessing frame
+        if self.precessing is True and self.frame == "inertial":
+            warnings.warn(
```

Comment:
yes, that came to my mind as well. However, the issue was that this was happening while processing the dataDict before setting values to debug_level. The debug_level is available via extra_kwargs which we set only after processing the dataDict.

Reply:
I see, was there a good reason for doing it like that? Would it make sense to flip things around, or even just read or pass extra_kwargs['debug_level'] where needed?

Reply:
Will the following work? Inside `process_dataDict` we check if `debug_level` is provided by the user via `extra_kwargs` and use it if its there else use the default value 0 as in the `default_extra_kwargs`

Reply:
The reason was that we needed information, for example, the width for peak finding, from the `process_dataDict` and then only set the default values.

Reply:
Replaced `warnings.warn` with `debug_message`

Reply:
Done!

Reply:
Hmm, don't know if that's the right approach...now there are two places where debug_level has defaults being set. Would it be better to just take it out of extra_kwargs?

Reply:
But if `extra_kwargs` is not provided then it's None which will raise an error.

Reply:
No, I mean, what is the entire debug_level option is made independent of extra_kwargs, like we did for https://github.com/vijayvarma392/gw_eccentricity/blob/main/gw_eccentricity/eccDefinition.py#L144. I don't know if this is the right approach, asking.

Reply:
Yes, that will solve this issue. Will need to make changes at several places though so that the checks pass. Would it be better to make that change in a separate PR?

Reply:
Sounds good. Revert the last commit first?

Reply:
Reverted.